libs/ledger-live-common/src/bridge/impl.ts

```diff
@@ -61,5 +69,53 @@ export function getAccountBridgeByFamily(family: string, accountId?: string): Ac
   if (!jsBridge) {
     throw new CurrencyNotSupported("currency bridge not found " + family);
   }
-  return jsBridge.accountBridge;
+  return wrapAccountBridge(jsBridge.accountBridge);
+}
+
+function wrapAccountBridge(bridge: AccountBridge<any>): AccountBridge<any> {
+  return {
+    ...bridge,
+    getTransactionStatus: async (...args) => {
+      const blockchainSpecific = await bridge.getTransactionStatus(...args);
+      const common = await commonGetTransactionStatus(...args);
+      const merged = mergeResults(blockchainSpecific, common);
+      return merged;
+    },
+  };
+}
+
+function mergeResults(
+  blockchainSpecific: TransactionStatusCommon,
+  common: Partial<TransactionStatusCommon>,
+): TransactionStatusCommon {
+  const errors = { ...blockchainSpecific.errors, ...common.errors };
+  const warnings = { ...blockchainSpecific.warnings, ...common.warnings };
+  return { ...blockchainSpecific, errors, warnings };
+}
+
+async function commonGetTransactionStatus(
+  account: Account,
+  transaction: any,
+): Promise<Partial<TransactionStatusCommon>> {
+  const errors: Record<string, Error> = {};
+  const warnings: Record<string, Error> = {};
+
+  let recipientIsBlacklisted = false;
+  if (transaction.recipient && transaction.recipient !== "") {
+    recipientIsBlacklisted = await isAddressSanctioned(account.currency, transaction.recipient);
+    if (recipientIsBlacklisted) {
+      errors.recipient = new RecipientAddressSanctionedError();
+    }
+  }
+
+  const userIsBlacklisted = await isAddressSanctioned(account.currency, account.freshAddress);
+  if (userIsBlacklisted) {
+    errors.amount = new UserAddressSanctionedError();
+  }
+
+  if (userIsBlacklisted || recipientIsBlacklisted) {
+    // Send log
+  }
+
+  return Promise.resolve({ errors, warnings });
```

Comment:
This is an `async` function
```suggestion
  return { errors, warnings };
```